### ISSUE TITLE
Douglas-Peucker algorithm in Luxor.simplify

### ISSUE THREAD

**Issue by pauljurczak** [JuliaGraphics/Luxor.jl#98]
I'm getting:
```
pol = [Point(0,0), Point(1,1), Point(2,0), Point(3,0), Point(4,0)]
.....
julia> simple = simplify(pol, 0.9)
3-element Array{Point,1}:
 Point(0.0, 0.0)
 Point(3.0, 0.0)
 Point(4.0, 0.0)
```
which is can be trivially simplified to `[Point(0,0), Point(4,0)]` and is not what Douglas-Peucker algorithm would produce, which is: `[Point(0,0), Point(1,1), Point(4,0)]`

**Comment by cormullion** [comment]
Do you think this an edge case or a more general problem with the algorithm...? Perhaps a set of test cases might be a good thing to have.

**Comment by pauljurczak** [comment]
This is a pretty simple test, not edgy at all. I may have a bit of time later today to look at the code and maybe find out what is going on.

**Comment by cormullion** [comment]
I think this might be an mistake in the `pointlinedistance()` function, which uses an odd triangle method... It will probably take a while to integrate a better version...

**Comment by pauljurczak** [comment]
It definitely is:
```
julia> pointlinedistance(Point(1, 1), Point(0, 0), Point(2, 0))
0.5
```
The distance should be 1. BTW, I'm going to implement Douglas-Peucker algorithm hopefully today. I will take a similar approach to yours, but I will store only a single index on the stack and I will optimize point to line distance computation. I can also fix `pointlinedistance` in the process, but I won't use it for `simplify`.

**Comment by cormullion** [comment]
Cool ! That would be awesome. Don’t forget to make a branch, then git pull/fetch so you’re working on a branch of master. Tests would be a bonus!

**Comment by pauljurczak** [comment]
I will add tests. Is there anything else I need to do with my pending pull request? Should I delete the branch after pull request is approved?

**Comment by cormullion** [comment]
Once the PR is merged, the branch is no longer required. I don’t believe they’re bulky or otherwise burdensome though.

**Comment by cormullion** [comment]
`pointlinedistance()` hopefully fixed by #101. Thanks!

**Comment by pauljurczak** [comment]
`pointlinedistance()` is definitely fixed now, `simplify()` will follow shortly.
